addons/hw_drivers/drivers/DisplayDriver.py

```diff
@@ -0,0 +1,204 @@
+from odoo.addons.hw_drivers.controllers.driver import Driver, event_manager, iot_devices
+
+path = os.path.realpath(os.path.join(os.path.dirname(__file__), '../views'))
+loader = jinja2.FileSystemLoader(path)
+
+jinja_env = jinja2.Environment(loader=loader, autoescape=True)
+jinja_env.filters["json"] = json.dumps
+
+pos_display_template = jinja_env.get_template('pos_display.html')
+
+_logger = logging.getLogger(__name__)
+
+
+class DisplayDriver(Driver):
+    connection_type = 'display'
+
+    def __init__(self, device):
+        super(DisplayDriver, self).__init__(device)
+        self._device_type = 'display'
+        self._device_connection = 'hdmi'
+        self._device_name = device['name']
+        self._connected = True
+        self.event_data = threading.Event()
+        self.owner = False
+        self.rendered_html = ''
+        self.load_url()
+
+    @property
+    def device_identifier(self):
+        return self.dev['identifier']
+
+    @classmethod
+    def supported(cls, device):
+        return True  # All devices with connection_type == 'display' are supported
+
+    @classmethod
+    def get_default_display(cls):
+        displays = list(filter(lambda d: iot_devices[d].device_type == 'display', iot_devices))
+        return len(displays) and iot_devices[displays[0]]
+
+    def action(self, data):
+        if data.get('action') == "update_url":
+            self.update_url(data.get('url'))
+        elif data.get('action') == "display_refresh":
+            self.call_xdotools('F5')
+        elif data.get('action') == "take_control":
+            self.take_control(self.data['owner'], data.get('html'))
+        elif data.get('action') == "customer_facing_display":
+            self.update_customer_facing_display(self.data['owner'], data.get('html'))
+        elif data.get('action') == "get_owner":
+            self.data = {
+                'value': '',
+                'owner': self.owner,
+            }
+            event_manager.device_changed(self)
+
+    def disconnect(self):
+        self._connected = False
+        super(DisplayDriver, self).disconnect()
+
+    def run(self):
+        while self._connected:
+            if self.url != 'http://localhost:8069/point_of_sale/display/':
+                # Refresh the page every minute
+                time.sleep(60)
+                self.call_xdotools('F5')
+
+    def update_url(self, url=None):
+        firefox_env = os.environ.copy()
+        firefox_env['HOME'] = '/tmp'
+        firefox_env['DISPLAY'] = ':0.0'
+        firefox_env['XAUTHORITY'] = '/run/lightdm/pi/xauthority'
+        self.url = url or 'http://localhost:8069/point_of_sale/display/'
+        if self.device_identifier != 'distant_display':
+            subprocess.Popen(['firefox', self.url], env=firefox_env)
+
+    def load_url(self):
+        if helpers.get_odoo_server_url():
+            result = requests.get("%s/iot/box/%s/screen_url" % (helpers.get_odoo_server_url(), helpers.get_mac_address()), {})
```

Comment:
do you try with pos in https and selfsigned certificate

Reply:
# disable certifiacte verification
            urllib3.disable_warnings()
            http = urllib3.PoolManager(cert_reqs='CERT_NONE')